modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/services/checker/excel/SpreadsheetFileService.scala

```diff
@@ -21,13 +21,25 @@ class SpreadsheetFileService {
     objectMapper.readValue(file, classOf[ExcelMediaInformationTasks])
   }
 
-  def getSubmissionFile(submissionID: Int): File = {
-    val submissionPath = this.storageService.pathToSolutionFile(submissionID).get.toString
-    new File(submissionPath)
+  def getSubmissionFile(submissionID: Int, cc: CheckrunnerConfiguration): File = {
+    if (cc.isInBlockStorage) {
+      val tmpFile = new File("solution-file")
+      storageService.getFileFromBucket("submissions", s"${cc.taskId}/solution-file", "solution-file")
+      tmpFile
+    } else {
+      val submissionPath = this.storageService.pathToSolutionFile(submissionID).get.toString
+      new File(submissionPath)
+    }
```

Comment:
I would move the If statment to the storage service so that you don't have to write it again and again.